quazip/JlCompress.cpp

```diff
@@ -76,6 +76,47 @@ bool JlCompress::compressFile(QuaZip* zip, QString fileName, QString fileDest) {
     return outFile.getZipError() == UNZ_OK;
 }
 
+bool JlCompress::compressFile(QuaZip* zip, QString fileName, QString fileDest, CompressionStrategy strategy)
```

Comment:
To avoid code duplication, wouldn't it be cleaner if `JlCompress::compressFile(QuaZip* zip, QString fileName, QString fileDest)` internally called `JlCompress::compressFile(QuaZip* zip, QString fileName, QString fileDest, CompressionStrategy strategy)` with default compression strategy?

Similar comment to other overloads.

Reply:
It's certainly feasible this way, and so far only I have tested the code successfully without encountering any issues. However, I'm concerned that when others start using it, unforeseen problems might arise.

Therefore, I'm debating whether to first utilize the current version for a period of time. If no issues surface during that interval, then we proceed with the modifications. Alternatively, we could modify the code immediately, and in case any problems emerge, we would revert back to the previous version.

Reply:
It should be pretty safe because lower level functions do the same thing. For example:

```
inline bool open(OpenMode mode, const char *password)
    {return open(mode, nullptr, nullptr, false, password);}
```

this just calls with method and level as nullptr and the same logic could be replicated here.

However, I do see your concern, how do we make absolutely sure we are still compressing with default method/level across all tests?

Perhaps one way would be to compute hashed of input and output files in current unit tests, hardcode and compare them inside the tests. Zip determinism is a bit tricky but it might be possible.